File: LeetCode/JavaScript/#415 Add Strings.js
/**
 * @param {string} num1
 * @param {string} num2
 * @return {string}
 */
var addStrings = function(num1, num2) {
    var len1 = num1.length;
    var len2 = num2.length;
    var flag = "0".charCodeAt()*2;
    var carry = 0;
    var tmp = Math.abs(len1-len2);
    var res= "";
    for(let i = 0;i<tmp;i++){
      if(len1<len2){
        num1 = "0"+num1;
      }else{
        num2 = "0"+num2;
      }
    }
    for(let i = num1.length-1;i>=0;i--){
      tmp = num1[i].charCodeAt()+num2[i].charCodeAt()-flag+carry;
      res = String(tmp%10)+res;
      carry =Math.floor(tmp/10);
    }
    res = carry===0?res:""+(carry+res);
    return res;
};

File: LeetCode/Python/#415 Add Strings.py
class Solution(object):
    def addStrings(self,num1, num2):
        """
        :type num1: str
        :type num2: str
        :rtype: str
        """
        flag = 48*2
        carry = 0
        tmp = abs(len(num1)-len(num2))
        res = ""
        for i in range(tmp):
            if len(num1)>len(num2):
                num2 = "0"+num2
            else:
                num1 = "0"+num1
        for i in range(len(num1)-1,-1,-1):
            tmp = ord(num1[i])+ord(num2[i]) - flag +carry
            res = str(tmp%10)+res;
            carry = tmp//10
        if not carry == 0:
            res = str(carry)+res
        return res

# addStrings("9333852702227987","85731737104263")